src/community/label_propagation.c

```diff
@@ -319,15 +322,18 @@ igraph_error_t igraph_community_label_propagation(const igraph_t *graph,
             }
 
             if (igraph_vector_int_size(&dominant_labels) > 0) {
-                /* Select randomly from the dominant labels */
-                k = RNG_INTEGER(0, igraph_vector_int_size(&dominant_labels) - 1);
-                k = VECTOR(dominant_labels)[k];
-                /* Check if the _current_ label of the node is also dominant */
-                if (VECTOR(label_counters)[(long)VECTOR(*membership)[v1]] != max_count) {
-                    /* Nope, we need at least one more iteration */
-                    running = 1;
+                if (control_iteration) {
+                    /* Check if the _current_ label of the node is also dominant */
+                    if (VECTOR(label_counters)[(long)VECTOR(*membership)[v1]] != max_count) {
```

Comment:
This cast to `long` should not have been there in the original code (it was a bug). Let's remove it now.

```suggestion
                    if (VECTOR(label_counters)[VECTOR(*membership)[v1]] != max_count) {
```